src/univariate/continuous/normal.jl

```diff
@@ -1,12 +1,18 @@
-immutable Normal <: ContinuousUnivariateDistribution
-    μ::Float64
-    σ::Float64
+immutable Normal{T<:Number} <: ContinuousUnivariateDistribution
+    μ::T
+    σ::T
 
-    Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); new(μ, σ))
-    Normal(μ::Real) = Normal(μ, 1.0)
-    Normal() = Normal(0.0, 1.0)
+    Normal(μ::T, σ::T) = new(μ, σ)
 end
 
+Normal{T<:Number}(μ::T, σ::T) = Normal{T}(μ, σ)
+Normal(μ::Number, σ::Number) = Normal(promote(μ, σ)...)
+Normal(μ::Number) = Normal(μ, one(μ))
+#Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); Normal(μ, σ))
+Normal(μ::Real) = Normal(μ, one(μ))
+Normal() = Normal(0.0, 1.0)
+Normal(μ::Complex, σ::Complex) = error("Normal with complex mean and variance not defined")
```

Comment:
I'm not entirely sure that saying that dual numbers are a special case of generalized complex numbers gives much insight into what to do here since the dual numbers are the only one of those generalizations that can be algebraically mapped back to the reals. It's pretty easy to see that if you're adjoining an element to ℝ and then identifying its square with some real value, the only value that allows you to recover ℝ by further modding is zero – i.e. the dual numbers.